app/src/main/java/io/neurolab/MainActivity.java

```diff
@@ -171,4 +194,90 @@ public boolean onNavigationItemSelected(@NonNull MenuItem item) {
         drawer.closeDrawer(GravityCompat.START);
         return true;
     }
+
+    private void changeTheme() {
+        int colorToBackground;
+        int colorFromBackground;
+        int colorFromStatusCardView;
+        int colorToStatusCardView;
+        int colorFromValues;
+        int colorToValues;
+
+        if (AppCompatDelegate.getDefaultNightMode() == AppCompatDelegate.MODE_NIGHT_YES) {
+            AppCompatDelegate.setDefaultNightMode(AppCompatDelegate.MODE_NIGHT_NO);
+            colorFromBackground = ContextCompat.getColor(this, R.color.colorPrimaryDark);
+            colorToBackground = ContextCompat.getColor(this, R.color.color_white_background);
+            colorFromStatusCardView = ContextCompat.getColor(this,R.color.colorCardViewNight);
+            colorToStatusCardView = ContextCompat.getColor(this,R.color.colorCardViewDay);
+            colorFromValues = ContextCompat.getColor(this,R.color.colorValuesNight);
+            colorToValues = ContextCompat.getColor(this,R.color.colorValuesDay);
+        } else {
+            AppCompatDelegate.setDefaultNightMode(AppCompatDelegate.MODE_NIGHT_YES);
+            colorFromBackground = ContextCompat.getColor(this, R.color.color_white_background);
+            colorToBackground = ContextCompat.getColor(this, R.color.colorPrimaryDark);
+            colorFromStatusCardView = ContextCompat.getColor(this, R.color.colorCardViewDay);
+            colorToStatusCardView = ContextCompat.getColor(this, R.color.colorCardViewNight);
+            colorFromValues = ContextCompat.getColor(this,R.color.colorValuesDay);
+            colorToValues = ContextCompat.getColor(this,R.color.colorValuesNight);
+        }
+
+        changeBackgroundColor(colorFromBackground,colorToBackground);
+        changeStatusCardViewColor(colorFromStatusCardView,colorToStatusCardView);
+        changeSetupCardViewColor(colorFromStatusCardView,colorToStatusCardView);
+        changeTextColor(colorFromValues,colorToValues);
+
+    }
+
+    private void changeBackgroundColor(int colorFrom, int colorTo) {
```

Comment:
All the functions are the same, create a single function and pass the view as the third argument

Reply:
Try to make a separate helper class that does the same

Reply:
understood, thank you.